### PR TITLE
wram: separate wOAMBuffer and wLinkOAMBuffer again

### PR DESCRIPTION
**Pull request by kemenaran**
The OAMBuffer area can be used for several purposes.

One is on the World gameplay mode, where indeed part of it is allocated for Link sprites, and part of it for entities.

But it is also used in other gameplay modes, like cutscenes, menus, etc to store a variety of arbitrary sprites.

So we need to keep both variables names (wOAMBuffer and wLinkOAMBuffer),
even if they refer to the same address technically.

In the code, sometimes the code purpose is clearly to update Links sprites (and in
this case wLinkOAMBuffer is better used). But otherwise, when the code is simply trying to manipulate arbitrary OAM data, wOAMBuffer is a better fit.

### CODE REVIEW

**Comment by zelosos** [comment]
But why not use two labels for one address?

File: src/constants/memory/wram.asm [hunk 1]
```diff
@@ -17,7 +17,17 @@ section "WRAM Bank0", wram0[$c000]
...
-; Buffer Area for 12 Link OAM tiles. Copied to OAM by DMA.
+; Generic 40 tiles buffer for OAM data. Copied to OAM by DMA.
+; During World gameplay, the OAM buffer is split between 12 OAM
+; tiles for Link, and 28 OAM tiles for dynamic entities.
+; Structure:
+;  byte 0: Y position
+;  byte 1: X position
+;  byte 2: tile n°
+;  byte 3: attribute
+wOAMBuffer:: ; C000
```
Comment (zelosos) [inline src/constants/memory/wram.asm:28]: remove the address after the label.
Addresses are noted after the ds X ;

File: src/constants/memory/wram.asm [hunk 2]
```diff
@@ -17,7 +17,17 @@ section "WRAM Bank0", wram0[$c000]
...
+
+; During World gameplay, buffer for 12 Link OAM tiles
 ; Structure:
 ;  byte 0 = Y position,
 ;  byte 1 = X position,
```
Comment (zelosos) [inline src/constants/memory/wram.asm:33]: ; add a comment for byte 2 -3 or a TODO comment if unknown yet

File: src/constants/memory/wram.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/memory/wram.asm]: Good catch 👍 

Fixed.

File: src/constants/memory/wram.asm (hunk 2 above)
Comment (kemenaran) [reply src/constants/memory/wram.asm:33]: The comments for bytes 2-3 is just below (Github is just not showing than many lines in the diff until an "Expand" icon is clicked in the gutter).

Comment (zelosos) [reply src/constants/memory/wram.asm:33]: ok

**Comment by zelosos** [comment]
looks good for me to merge

**Comment by kemenaran** [comment]
Thanks for the review 🙇

**Pull request merged**